protobuf/types/service.proto

```diff
@@ -54,7 +54,48 @@ message Service {
     string command = 5;               // Command to run the container.
     repeated string env = 9;          // Default env vars to apply to dependency on runtime.
   }
-  
+
+  // A workflow is a configuration to trigger a specific task when certains conditions of a trigger are valid.
+  message Workflow {
+
+    // Trigger is the configuration that will trigger a workflow.
+    message Trigger {
+
+      // Type of trigger available. Event to react to an event. Result to react to a result of an execution.
+      enum TriggerType {
+        Event = 0;  // React to event.
+        Result = 1; // React to result of an execution.
+      }
+
+      // Filter is applied on the data of the event/result.
+      message Filter {
+
+        // Type of condition available to compare the values.
+        enum Predicate {
+          EQ = 0; // Equal
+        }
+
+        string key = 1;           // Key to check.
+        Predicate predicate = 2;  // Type of condition to apply.
+        string value = 3;         // Value of the filter
+      }
+
+      TriggerType type = 1;         // Type of trigger.
+      string instanceHash = 2;      // Hash of the instance that triggers the workflow.
+      string key = 3;               // Key that triggers the workflow (event key or task key).
+      repeated Filter filters = 4;  // List of filters to apply on the data of the event/result.
```

Comment:
Filters are really important if we want to do workflows that are useful. Also, this is developed already, the part here is really the integration in the API with protobuf that doesn't support any non-typed random data.

The filter definitely needs more work. I will write something on the workflow post that could describe the structure of the filter and open the discussion to improve it. But except if there any good reason, I don't see the point to remove it now. It's just a first version of the filter that will have new features but is still valid.

Reply:
I'm ok to keep it for test but we have to standardize filters so they can be used on anything before any official release.
Also, a filter can be done by a task of a service that will filter the inputs (maybe with the JS service)

Reply:
I was thinking about a filter with another service but in that case, we lose the access of the initial data so this doesn't work.
Also, I'm totally fine to have this generic filter system but let's not block anything because of that. We already have different filters in the codebase right now (execution stream, event stream) and it didn't block any release.

Reply:
For the filter, there is already a post on the forum https://forum.mesg.com/t/filter-events-and-results-by-data/149

Reply:
Following might give some insights:
https://github.com/ilgooz/service-logic
https://github.com/ilgooz/service-objects